### PR TITLE
Replication case

### PR DESCRIPTION
**Pull request by abhayayay**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

-  Added Delete user API call for sync gateway
- API update for creating users to incorporate collection access.
- Replication test cases for cbl-collections
-  Conftest file to avoid reset cluster in cbl replication collections.
- P2P function call methods.
- The test cases are compatible with cbl android/java and ios. Compatibility with C is in progress.

### CODE REVIEW

File: keywords/MobileRestClient.py [hunk 1]
```diff
@@ -377,7 +377,23 @@ def get_users(self, url, db, auth=None):
 
         return resp.json()
 
-    def create_user(self, url, db, name, password, channels=None, roles=None, auth=None):
```
Comment (gilkal) [inline keywords/MobileRestClient.py:380]: Let's get @muntazim-aliCB view on this, he's been working in that area as well.

File: CBLClient/PeerToPeer.py [hunk 2]
```diff
@@ -93,6 +95,37 @@ def server_stop(self, url_listener, end_point_type):
         args.setString("endPointType", end_point_type)
         return self._client.invokeMethod("peerToPeer_serverStop", args)
 
+    def configureCollection(self, host, server_db_name, port=5000, continuous=None, authenticator=None, replication_type=None, endPointType="MessageEndPoint", basic_auth=None,
```
Comment (gilkal) [inline CBLClient/PeerToPeer.py:98]: This is quite a big list of arguments. Cld this list be reduced? A big list of arguments can cause conflicting argument behaviour, amongst others. 

Possible options to reduce could include splitting one general function to smaller functions and remove some of the arguments that are not currently being used.

File: keywords/MobileRestClient.py [hunk 3]
```diff
@@ -388,15 +404,18 @@ def create_user(self, url, db, name, password, channels=None, roles=None, auth=N
         if roles is None:
             roles = []
 
-        types.verify_is_list(channels)
-        types.verify_is_list(roles)
-
         data = {
             "name": name,
             "password": password,
-            "admin_channels": channels,
```
Comment (muntazim-aliCB) [inline keywords/MobileRestClient.py:397]: It is possible to have both a top level `admin_channels` and those under `collection_access` for backwards compatibility; the top level maps to the `_default._default` collection

**Review by muntazim-aliCB — CHANGES_REQUESTED** [review]

File: keywords/MobileRestClient.py [hunk 4]
```diff
@@ -421,12 +440,15 @@ def update_user(self, url, db, name, password=None, channels=None, roles=None, d
 
         types.verify_is_list(channels)
         types.verify_is_list(roles)
-
         data = {
```
Comment (muntazim-aliCB) [inline keywords/MobileRestClient.py:443]: Can you bring this into line with changes mentioned in `create_user`?

File: keywords/MobileRestClient.py [hunk 5]
```diff
@@ -377,7 +377,23 @@ def get_users(self, url, db, auth=None):
...
+
+    def delete_user(self, url, db, name, auth=None):
+        if auth:
+            resp = self._session.delete("{}/{}/_user/{}".format(url, db, name), auth=HTTPBasicAuth(auth[0], auth[1]))
+        else:
+            resp = self._session.delete("{}/{}/_user/{}".format(url, db, name))
+        log_r(resp)
+        resp.raise_for_status()
+
+    def create_user(self, url, db, name, password, channels=None, roles=[], auth=None, scope_dict=None):
```
Comment (muntazim-aliCB) [inline keywords/MobileRestClient.py:396]: For comprehension can you rename the `scope_dict` argument to `collection_access`, as that is what it is defining

File: CBLClient/PeerToPeer.py (hunk 2 above)
Comment (abhayayay) [reply CBLClient/PeerToPeer.py]: I have removed the parameters that are not needed at this point of time.

**Review by muntazim-aliCB — APPROVED** [review]

**Review by gilkal — APPROVED** [review]

**Pull request merged**